spec/services/error_details/create_service_spec.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ErrorDetails::CreateService, type: :service do
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:customer) { create(:customer, organization:) }
+  let(:integration) { create(:anrok_integration, organization:) }
+  let(:owner) { create(:invoice, organization:, customer:) }
+
+  describe '#call' do
+    subject(:service_call) { described_class.call(params:, integration:, owner:, organization:) }
+
+    let(:params) do
+      {
+        error_code: 'not_provided',
+        details: {'error_code' => 'taxDateTooFarInFuture'}
```

Comment:
I am curious how did you design it? It this case we will receive `taxDateTooFarInFuture` from the Anrok but shouldn't we save it as `error_code`? If not, what should we set there?

Reply:
originally I thought to store everything in the details and not have this separate error_code.
But then I saw that when having more generic error that can store any errors, we might want to query some specific errors for an organization, and querying by jsonb field will not be that convenient. 
so the error_code got extracted to the model level
but then 
1) we might receive more information about the error that can be stored in the jsonb field
2) I was thinking we can decide it when creating the error, but specifically for anrok we can set error_code = 'tax_error' (or just keep the error_code in the error_code and keep details empty)